File: studio/src/main/java/br/com/api/enums/TypeOfCurrencyEnum.java
package br.com.api.enums;

import lombok.AllArgsConstructor;
import lombok.Getter;

@AllArgsConstructor
@Getter
public enum TypeOfCurrencyEnum {

	CASH("label.cash"),
	CHECK("label.check"),
	DEBIT_CARD("label.debit.card"),
	CREDIT_CARD("label.credit.card"),
	CREDIT("label.credit"),
	TRADE_NOTE("label.trade.note"),
	BONUS("label.bonus"),
	PIX("label.pix");

	private final String i18n;
}

File: src/model/TypeOfCurrencyEnum.ts
export enum TypeOfCurrencyEnum {
    CASH = 'Dinheiro',
    CHECK = 'Cheque',
    DEBIT_CARD = 'Cartão de débito',
    CREDIT_CARD = 'Cartão de crédito',
    CREDIT = 'Crédito',
    TRADE_NOTE = 'Boleto',
    BONUS = 'Bonus',
    PIX = 'Pix'
}

export const TypeOfCurrencyList: {
    key: string;
    value: string;
}[] = Object.entries(TypeOfCurrencyEnum).map(([key, value]) => ({ key, value }))

export type TypeOfCurrencyType = keyof typeof TypeOfCurrencyEnum

export const TypeOfCurrencyEntries = Object.entries(TypeOfCurrencyEnum)
